### PR TITLE
Fix zoo export

### PR DESCRIPTION
**Pull request by KSGulin**
This PR removes `anchor_grid` from the state dict/ unpickled model if it is found. Legacy sparsezoo models (v5.0) are saved with `anchor_grid`. This causes issues as `anchor_grid` processing has been updated between versions and the legacy `anchor_grid` are not in the correct format to be processed successfully. The values for `anchor_grid` are determined from the `Detect` layer's `anchors` on runtime and are not needed for model loading/saving.

**Manual Testing**
`sparseml.yolov5.export_onnx --weights zoo:cv/detection/yolov5-l/pytorch/ultralytics/coco/pruned-aggressive_98`
 
`sparseml.yolov5.export_onnx --weights zoo:cv/detection/yolov5-s/pytorch/ultralytics/coco/pruned_quant-aggressive_94`

```
sparseml.yolov5.train --cfg models/yolov5s.yaml --epochs 1
sparseml.yolov5.export_onnx --weights yolov5_runs/train/exp/weights/last.pt
```

```
sparseml.yolov5.train --cfg models_v5.0/yolov5s.yaml --epochs 1
sparseml.yolov5.export_onnx --weights yolov5_runs/train/exp/weights/last.pt
```

```
sparseml.yolov5.train --one-shot --cfg models_v5.0/yolov5l.yaml --recipe zoo:cv/detection/yolov5-l/pytorch/ultralytics/coco/pruned-aggressive_98
sparseml.yolov5.export_onnx --weights yolov5_runs/train/exp/weights/checkpoint-one-shot.pt
```

```
sparseml.yolov5.train --one-shot --cfg models_v5.0/yolov5l.yaml --recipe zoo:cv/detection/yolov5-s/pytorch/ultralytics/coco/pruned_quant-aggressive_94
sparseml.yolov5.export_onnx --weights yolov5_runs/train/exp/weights/checkpoint-one-shot.pt
```

**Almost Automatic Testing**
`make testinteg TARGETS=yolov5`

`SPARSEML_TEST_CADENCE='commit' make testinteg TARGETS=yolov5`

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def train(hyp,  # path/to/hyp.yaml or hyp dictionary
         loggers.on_params_update({"batch_size": batch_size})
 
     # Optimizer
-    nbs = 64  # nominal batch size
+    nbs = batch_size*opt.gradient_accum_steps if opt.overwrite_batch_size else 64   # nominal batch size
```
Comment (corey-nm) [inline train.py:169]: Are the changes to this file supposed to be in this PR? They seem unrelated to anchor_grid changes

Comment (KSGulin) [reply train.py]: Fixed, thanks for catching that. Those changes are a part of the open PR #90

File: models/experimental.py [hunk 2]
```diff
@@ -95,6 +95,13 @@ def attempt_load(weights, map_location=None, inplace=True, fuse=True):
     for w in weights if isinstance(weights, list) else [weights]:
         ckpt = torch.load(attempt_download(w), map_location=map_location)  # load
         ckpt = (ckpt.get('ema') or ckpt['model']).float()  # FP32 model
+
+        # backwards compatability for sparsezoo models that maintain anchor grid in state dict
+        anchor_grid_key = [key for key in ckpt.state_dict().keys() if "anchor_grid" in key]
+        if len(anchor_grid_key) == 1:
+            del ckpt.model[-1].anchor_grid
+            ckpt.model[-1].anchor_grid = [torch.zeros(1)] * ckpt.model[-1].nl # reset to init value
```
Comment (corey-nm) [inline models/experimental.py:103]: So in this file you aren't actually removing it from state_dict, but replacing? Whereas other file you are just removing from state dict? How come they are different?

Comment (corey-nm) [reply models/experimental.py:103]: Oh i see in the PR description that export just doesn't need anchor_grid? Is this experimental.py script part of the training stuff that needs anchor grid in the new format?

Comment (KSGulin) [reply models/experimental.py:103]: This fix is highlighting the issue/complexity that comes with having 2+ load flows. Yolov5 typically saves the full model in pickled format rather than just the state dict. We had to add a state dict path because pickling isn't supported for quantized models. So now we have a path for each. 

**Load state dict fix**
- Model is initialized from scratch (and thus has the correct init anchor_grid) _[usual behavior]_
- `anchor_grid` is removed from the state dict _[added with fix]_
- State dict is loaded into model _[usual behavior]_

**Load pickled model fix**
- Pickled file is loaded _[usual behavior]_
- `anchor_grid` field is re-initialized _[added with fix]_
- Model is as the actual model, no state dict application _[usual behavior]_

Comment (KSGulin) [reply models/experimental.py:103]: To clarify, for non-quantized models we still use the original(ish) path

Comment (corey-nm) [reply models/experimental.py:103]: I see yeah that is complicated. thanks for the details!

**Review by corey-nm — APPROVED** [review]

File: export.py [hunk 3]
```diff
@@ -519,6 +519,12 @@ def load_checkpoint(
                       anchors=hyp.get('anchors') if hyp else None).to(device)
         model_key = 'ema' if (not train_type and 'ema' in ckpt and ckpt['ema']) else 'model'
         state_dict = ckpt[model_key].float().state_dict() if pickled else ckpt[model_key]
+
+        # backwards compatability for sparsezoo models that maintain anchor grid in state dict
```
Comment (kylesayrs) [inline export.py:523]: 🎉

**Review by kylesayrs — APPROVED** [review]
